src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsFeatureService.cs

```diff
@@ -237,8 +237,22 @@ private static async Task<Document> CleanUpNewLinesAsync(Document document, Text
             IDocumentTextDifferencingService textDiffingService,
             CancellationToken cancellationToken)
         {
-            var newSolution = await codeAction.GetChangedSolutionAsync(
-                progressTracker, cancellationToken: cancellationToken).ConfigureAwait(false);
+            var operations = await codeAction.GetOperationsAsync(cancellationToken).ConfigureAwait(false);
+            Solution newSolution;
+            if (operations.Length == 0)
+            {
+                newSolution = document.Project.Solution;
+            }
+            else if (operations.Length == 1 && operations[0] is ApplyChangesOperation applyChangesOperation)
+            {
+                newSolution = applyChangesOperation.ChangedSolution;
```

Comment:
why do we need these two branches unrolled here?  wouldn't GetChangedSolutionAsync just do this?

Reply:
(and, if it doesn't, then please doc why we're special casing tehse two cases here).

Reply:
➡️ Documented both changes in this commit